cli/pkg/state/vmclarity.go

```diff
@@ -137,6 +137,26 @@ func (v *VMClarityState) MarkDone(ctx context.Context, errors []error) error {
 	return nil
 }
 
+func (v VMClarityState) IsAborted(ctx context.Context) (bool, error) {
+	scanResult, err := v.client.GetScanResult(ctx, v.scanResultID, models.GetScanResultsScanResultIDParams{
+		Select: utils.PointerTo("id,status"),
```

Comment:
why we are not doing filter on ABORTED state with the odata?

Reply:
The `models.GetScanResultsScanResultIDParams` does not support setting `filter` field for `GetScanResult`.

Using `GetScanResults` would allow filtering based on status like this: `/api/scanResults?$filter=id eq 'SCAN_RESULT_ID' and status/general/state eq 'ABORTED'`. However  the API returns and empty list if there is no `ScanResult`which matches the filter. The problem with this approach is that empty list could mean two things: either the `ScanResult` does not exist at all or the `ScanResult` does exist but has different state than ABORT. We need to be able to differentiate between the two cases as former must raise an error while the later means the `ScanResult` in question is in `ABORTED` state.

So I think the current implementation is the right way of checking whether the `ScanResult` is aborted or not.

Reply:
> The models.GetScanResultsScanResultIDParams does not support setting filter field for GetScanResult.

I think that this is just a missing implementation, why wouldn't we want to support it?

Reply:
I guess we can close it for now.